cmd/soroban-cli/src/commands/contract/deploy/asset.rs

```diff
@@ -118,7 +123,9 @@ impl NetworkRunnable for Cmd {
             data::write(get_txn_resp, &network.rpc_uri()?)?;
         }
 
-        Ok(stellar_strkey::Contract(contract_id.0).to_string())
+        Ok(TxnResult::Xdr(
+            stellar_strkey::Contract(contract_id.0).to_string(),
```

Comment:
Why is the contract address being wrapped in a TxnResult::Xdr? The fact that strings are convertible to XDR seems a bit odd for this use case. Contract addresses are not normally encoded into XDR.

Reply:
This should be a `TxnResult::Res` as this should return a strkey and you're right it shouldn't be encoded as a String.